client/signup/config/steps-pure.js

```diff
@@ -209,6 +209,18 @@ export function generateSteps( {
 			fulfilledStepCallback: isPlanFulfilled,
 		},
 
+		'plans-hosting': {
+			stepName: 'plans',
+			apiRequestFunction: addPlanToCart,
+			dependencies: [ 'siteSlug' ],
+			optionalDependencies: [ 'emailItem', 'themeSlugWithRepo' ],
+			providesDependencies: [ 'cartItem', 'themeSlugWithRepo' ],
+			fulfilledStepCallback: isPlanFulfilled,
+			props: {
+				hideFreePlan: true,
+			},
```

Comment:
I think this is a much better solution than hooking into the `flowName` inside `plans/index.jsx`.

The downside is that it renames the plans page from `/start/hosting/plans` to `/start/hosting/plans-hosting`. But I guess this is a valid trade-off.